piptools/utils.py

```diff
@@ -343,8 +332,6 @@ def get_compile_command(click_ctx):
                     # Instead, we try to get more legible quoting via repr:
                     left_args.append(f"{option_long_name}={repr(val)}")
                 else:
-                    left_args.append(
-                        f"{option_long_name}={shlex.quote(force_text(val))}"
-                    )
+                    left_args.append(f"{option_long_name}={shlex.quote(str(val))}")
```

Comment:
```suggestion
                    left_args.append(f"{option_long_name}={shlex.quote(val)}")
```

Extra `str`?

Reply:
That particular `str()` is necessary as `val` may not already be a string. It may be an int or other kind of CLI argument. Dropping it results in the following test failure:

```
____________________________ test_get_compile_command[cli_args13-pip-compile --max-rounds=42] ____________________________

tmpdir_cwd = local('/tmp/pytest-of-jon/pytest-6/test_get_compile_command_cli_a13'), cli_args = []
expected_command = 'pip-compile --max-rounds=42'

    @pytest.mark.parametrize(
        ("cli_args", "expected_command"),
        (
            # Check empty args
            ([], "pip-compile"),
            # Check all options which will be excluded from command
            (["-v"], "pip-compile"),
            (["--verbose"], "pip-compile"),
            (["-n"], "pip-compile"),
            (["--dry-run"], "pip-compile"),
            (["-q"], "pip-compile"),
            (["--quiet"], "pip-compile"),
            (["-r"], "pip-compile"),
            (["--rebuild"], "pip-compile"),
            (["-U"], "pip-compile"),
            (["--upgrade"], "pip-compile"),
            (["-P", "django"], "pip-compile"),
            (["--upgrade-package", "django"], "pip-compile"),
            # Check options
            (["--max-rounds", "42"], "pip-compile --max-rounds=42"),
            (["--index-url", "https://foo"], "pip-compile --index-url=https://foo"),
            # Check that short options will be expanded to long options
            (["-p"], "pip-compile --pre"),
            (["-f", "links"], "pip-compile --find-links=links"),
            (["-i", "https://foo"], "pip-compile --index-url=https://foo"),
            # Check positive flags
            (["--generate-hashes"], "pip-compile --generate-hashes"),
            (["--pre"], "pip-compile --pre"),
            (["--allow-unsafe"], "pip-compile --allow-unsafe"),
            # Check negative flags
            (["--no-index"], "pip-compile --no-index"),
            (["--no-emit-index-url"], "pip-compile --no-emit-index-url"),
            (["--no-emit-trusted-host"], "pip-compile --no-emit-trusted-host"),
            (["--no-annotate"], "pip-compile --no-annotate"),
            (["--no-allow-unsafe"], "pip-compile"),
            # Check that default values will be removed from the command
            (["--emit-trusted-host"], "pip-compile"),
            (["--annotate"], "pip-compile"),
            (["--index"], "pip-compile"),
            (["--emit-index-url"], "pip-compile"),
            (["--max-rounds=10"], "pip-compile"),
            (["--build-isolation"], "pip-compile"),
            # Check options with multiple values
            (
                ["--find-links", "links1", "--find-links", "links2"],
                "pip-compile --find-links=links1 --find-links=links2",
            ),
            # Check that option values will be quoted
            (["-f", "foo;bar"], "pip-compile --find-links='foo;bar'"),
            (["-f", "συνδέσεις"], "pip-compile --find-links='συνδέσεις'"),
            (["-o", "my file.txt"], "pip-compile --output-file='my file.txt'"),
            (["-o", "απαιτήσεις.txt"], "pip-compile --output-file='απαιτήσεις.txt'"),
            # Check '--pip-args' (forwarded) arguments
            (
                ["--pip-args", "--disable-pip-version-check"],
                "pip-compile --pip-args='--disable-pip-version-check'",
            ),
            (
                ["--pip-args", "--disable-pip-version-check --isolated"],
                "pip-compile --pip-args='--disable-pip-version-check --isolated'",
            ),
            pytest.param(
                ["--extra-index-url", "https://username:password@example.com/"],
                "pip-compile --extra-index-url='https://username:****@example.com/'",
                id="redact password in index",
            ),
            pytest.param(
                ["--find-links", "https://username:password@example.com/"],
                "pip-compile --find-links='https://username:****@example.com/'",
                id="redact password in link",
            ),
        ),
    )
    def test_get_compile_command(tmpdir_cwd, cli_args, expected_command):
        """
        Test general scenarios for the get_compile_command function.
        """
        with compile_cli.make_context("pip-compile", cli_args) as ctx:
>           assert get_compile_command(ctx) == expected_command

/home/jon/devel/pip-tools/tests/test_utils.py:283: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/jon/devel/pip-tools/piptools/utils.py:335: in get_compile_command
    left_args.append(f"{option_long_name}={shlex.quote(val)}")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

s = 42

    def quote(s):
        """Return a shell-escaped version of the string *s*."""
        if not s:
            return "''"
>       if _find_unsafe(s) is None:
E       TypeError: expected string or bytes-like object

/usr/lib64/python3.9/shlex.py:329: TypeError
```

Reply:
Oh, I see. Thanks!